plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/statistics.kt

```diff
@@ -0,0 +1,124 @@
+import org.jetbrains.kotlinx.dataframe.plugin.impl.dataFrame
+import org.jetbrains.kotlinx.dataframe.plugin.impl.simpleColumnOf
+import kotlin.reflect.KType
+import kotlin.reflect.full.createType
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.fir.types.ConeClassLikeType
+import org.jetbrains.kotlin.fir.types.ConeNullability
+import org.jetbrains.kotlin.fir.types.impl.ConeClassLikeTypeImpl
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.aggregators.Aggregator
+import org.jetbrains.kotlinx.dataframe.plugin.impl.SimpleCol
+import kotlin.reflect.KClass
+
+private object PrimitiveClassIds {
+    const val INT = "kotlin/Int"
+    const val LONG = "kotlin/Long"
+    const val DOUBLE = "kotlin/Double"
+    const val FLOAT = "kotlin/Float"
+    const val SHORT = "kotlin/Short"
+    const val BYTE = "kotlin/Byte"
+}
+
+private fun KClass<*>.toClassId(): ClassId? = when (this) {
+    Int::class -> ClassId.fromString(PrimitiveClassIds.INT)
+    Long::class -> ClassId.fromString(PrimitiveClassIds.LONG)
+    Double::class -> ClassId.fromString(PrimitiveClassIds.DOUBLE)
+    Float::class -> ClassId.fromString(PrimitiveClassIds.FLOAT)
+    Short::class -> ClassId.fromString(PrimitiveClassIds.SHORT)
+    Byte::class -> ClassId.fromString(PrimitiveClassIds.BYTE)
+    else -> null
+}
+
+private val primitiveTypeMap = mapOf(
+    PrimitiveClassIds.INT to Int::class,
+    PrimitiveClassIds.LONG to Long::class,
+    PrimitiveClassIds.DOUBLE to Double::class,
+    PrimitiveClassIds.FLOAT to Float::class,
+    PrimitiveClassIds.SHORT to Short::class,
+    PrimitiveClassIds.BYTE to Byte::class
+)
+
+fun ConeKotlinType.toKType(): KType? {
+    return (this as? ConeClassLikeType)?.let { coneType ->
+        val nullable = coneType.nullability == ConeNullability.NULLABLE
+        primitiveTypeMap[coneType.lookupTag.classId.asString()]
+            ?.createType(nullable = nullable)
+    }
+}
+
+fun KType.toConeKotlinType(): ConeKotlinType? {
+    val kClass = this.classifier as? KClass<*> ?: return null
+    val classId = kClass.toClassId() ?: return null
+
+    return ConeClassLikeTypeImpl(
+        lookupTag = ConeClassLikeErrorLookupTag(classId),
+        typeArguments = emptyArray(),
+        isNullable = this.isMarkedNullable
+    )
+}
+
+private fun Arguments.generateStatisticResultColumns(
+    statisticAggregator: Aggregator<Number, Number>,
+    inputColumns: List<SimpleDataColumn>
+): List<SimpleCol> {
+    return inputColumns.map { col -> createUpdatedColumn(col, statisticAggregator) }
+}
+
+private fun Arguments.createUpdatedColumn(
+    column: SimpleDataColumn,
+    statisticAggregator: Aggregator<Number, Number>
+): SimpleCol {
+    val originalType = column.type.type
+    val inputKType = originalType.toKType()
+    val resultKType = inputKType?.let { statisticAggregator.calculateReturnType(it, emptyInput = false) }
```

Comment:
might cause issues for min/max/median/percentile to set `emptyInput = false`. I'd set it to `true`. The name `emptyInput` is also kind of a misnomer, it should be "inputPotentiallyEmpty" or something.
It changes `intCol.min()` from `Int` (when it's set to false) to `Int?` (when it's set to true).
Feel free to rename the argument everywhere it's used :)